.github/workflows/upgrade-test.yaml

```
@@ -0,0 +1,177 @@
+            echo "Database file not found!"
+            exit 1
+          fi
+      
+      - name: Stop old HomeBox instance
+        run: |
+          docker stop homebox-old
+          docker rm homebox-old
+      
+      # Step 3: Build latest version from main branch
+      - name: Build HomeBox from main branch
+        run: |
```

actionlint (1.7.9)

shellcheck reported issue in this script: SC2046:warning:4:26: Quote this to prevent word splitting

(shellcheck)